pkg/operator/operator.go

```diff
@@ -153,7 +153,7 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		*sess.Config.Region,
 	)
 	versionProvider := version.NewDefaultProvider(operator.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval))
-	ssmProvider := ssmp.NewDefaultProvider(ssm.NewFromConfig(cfg), cache.New(awscache.SSMGetParametersByPathTTL, awscache.DefaultCleanupInterval))
+	ssmProvider := ssmp.NewDefaultProvider(ssm.NewFromConfig(cfg), cache.New(awscache.SSMGetParametersByPathTTL, awscache.DefaultCleanupInterval), operator.Clock, clusterEndpoint)
```

Comment:
Seems odd to me that we would pass clusterEndoint through here -- I ask myself why clusterEndpoint matters in the context of the SSM provider

Reply:
Yeah, I was thinking of moving this to the nodeclass status provider, still a strange dependency but necessary until we have the cluster provider I mentioned in another comment.